restlight-core/src/main/java/io/esastack/restlight/core/resolver/nav/NameAndValueResolver.java

```diff
@@ -0,0 +1,99 @@
+package io.esastack.restlight.core.resolver.nav;
+
+import esa.commons.Checks;
+import esa.commons.ObjectUtils;
+import io.esastack.restlight.core.context.RequestContext;
+import io.esastack.restlight.core.method.Param;
+import io.esastack.restlight.core.resolver.ParamResolver;
+import io.esastack.restlight.core.util.ConverterUtils;
+import io.esastack.restlight.server.bootstrap.WebServerException;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public class NameAndValueResolver<T> implements ParamResolver {
+
+    private final NameAndValue nav;
+    private final Converter<T> converter;
+    private final BiFunction<String, RequestContext, T> valueProvider;
+
+    public NameAndValueResolver(Param param,
+                                Converter<T> converter,
+                                BiFunction<String, RequestContext, T> valueProvider,
+                                Function<Param, NameAndValue> nameAndValueCreator) {
+        this.converter = Checks.checkNotNull(converter, "converter");
+        this.valueProvider = Checks.checkNotNull(valueProvider, "valueProvider");
+        this.nav = getNameAndValue(param,
+                Checks.checkNotNull(nameAndValueCreator, "nameAndValueCreator")
+                        .apply(Checks.checkNotNull(param, "param")));
+    }
+
+    @Override
+    public Object resolve(Param param, RequestContext ctx) {
+        Object arg = converter.convert(nav.name, ctx, valueProvider);
```

Comment:
Duplicate code segment, see more from `AbstractNameAndValueResolver#updatedNamedValue` and what's
the different between `AbstractNameAndValueResolver` and `NameAndValueResolver`?